src/fixate/drivers/pps/siglent_spd_3303X.py

```diff
@@ -206,11 +205,10 @@ def _write(self, data):
 
     def _check_errors(self):
         resp = self.instrument.query("SYST:ERR?")
-        comp = re.compile(r"(\d+) {1,2}([\w ]+)", re.UNICODE)
+        comp = re.compile(r"([+-]?\d+)\,? {1,2}([\w ]+)", re.UNICODE)
         resp = comp.match(resp)
```

Comment:
Would be good a have a few example strings you're trying to match against.
Or even better - a couple of unittests. If you add unit tests, probably easy to move the query out of the function.

Actually, it might even be worth cleaning up _is_error, since it is never called with the silent keyword arg. Just get rid of that logic altogether and simplify everything?

Reply:
Done all